build/gulp/tasks/test-projects.tsx

```diff
@@ -199,7 +202,45 @@ task('test:projects:rollup', async () => {
   logger(`✔️Browser test was passed`)
 })
 
+task('test:projects:typings', async () => {
+  const logger = log('test:projects:typings')
+
+  const scaffoldPath = paths.base.bind(null, 'build/gulp/tasks/test-projects/typings')
+  const tmpDirectory = tmp.dirSync({ prefix: 'stardust-' }).name
+
+  logger(`✔️Temporary directory was created: ${tmpDirectory}`)
+
+  const dependencies = [
+    '@types/react',
+    '@types/react-dom',
+    'react',
+    'react-dom',
+    'typescript',
+  ].join(' ')
+  await runIn(tmpDirectory)(`yarn add ${dependencies}`)
+  logger(`✔️Dependencies were installed`)
+
+  const packedPackages = await packStardustPackages(logger)
+  await addResolutionPathsForStardustPackages(tmpDirectory, packedPackages)
+  await runIn(tmpDirectory)(`yarn add ${packedPackages['@stardust-ui/react']}`)
+  logger(`✔️Stardust UI packages were added to dependencies`)
+
+  fs.mkdirSync(path.resolve(tmpDirectory, 'src'))
+  fs.copyFileSync(scaffoldPath('index.tsx'), path.resolve(tmpDirectory, 'src/index.tsx'))
+  fs.copyFileSync(scaffoldPath('tsconfig.json'), path.resolve(tmpDirectory, 'tsconfig.json'))
+  logger(`✔️Source and configs were copied`)
+
+  await runIn(tmpDirectory)(`yarn tsc --noEmit`)
+  logger(`✔️Example project was successfully built: ${tmpDirectory}`)
+})
+
 task(
   'test:projects',
-  series('dll', 'bundle:all-packages', parallel('test:projects:cra-ts', 'test:projects:rollup')),
+  series(
+    'dll',
+    'bundle:all-packages',
+    'test:projects:cra-ts',
+    'test:projects:rollup',
+    'test:projects:typings',
```

Comment:
We can't run in parallel now, `yarn` fails with cache errors, see previous builds